tests/CanadaPost/Tests/NCShipmentTest.php

```diff
@@ -0,0 +1,313 @@
+     *
+     * @covers NCShipment::getNCShipments()
+     * @test
+     */
+    public function getNCShipments()
+    {
+        $body = file_get_contents(__DIR__
+            . '/../Mocks/ncshipments-response-success.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $body),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $shipments = $this->shipmentService->getNCShipments('', '', '', ['handler' => $handler]);
+
+        // Check response.
+        $this->assertTrue(is_array($shipments['non-contract-shipments']));
+
+        // Test the shipments.
+        $links = $shipments['non-contract-shipments']['link'];
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787234',
+            $links[0]['@attributes']['href']);
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787352',
+            $links[1]['@attributes']['href']);
+
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787123',
+            $links[2]['@attributes']['href']);
+    }
+
+    /**
+     * Test the POST request to refund a shipment Canada Post.
+     *
+     * @covers NCShipment::requestNCShipmentRefund()
+     * @test
+     */
+    public function requestNCShipmentRefund()
+    {
+        $body = file_get_contents(__DIR__
+            . '/../Mocks/ncshipment-refund-request-info.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $body),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $refund = $this->shipmentService->requestNCShipmentRefund('', '', ['handler' => $handler]);
+
+        // Check response.
+        $this->assertTrue(is_array($refund['non-contract-shipment-refund-request-info']));
+
+        $this->assertEquals('2018-08-28', $refund['non-contract-shipment-refund-request-info']['service-ticket-date']);
+        $this->assertEquals('GT12345678RT', $refund['non-contract-shipment-refund-request-info']['service-ticket-id']);
+
+    }
+
+    protected function checkShippingResponse($shipment)
+    {
+        // Check response.
+        $this->assertTrue(is_array($shipment['non-contract-shipment-info']));
+
+        // Test the rates.
+        $links = $shipment['non-contract-shipment-info']['links']['link'];
+        $this->assertTrue(is_array($links));
+        $this->assertEquals('406951321983787352',
+            $shipment['non-contract-shipment-info']['shipment-id']);
+
+        $this->assertTrue(is_array($links));
+
+        // Self.
+        $this->assertEquals('https://ct.soa-gw.canadapost.ca/rs/0007023211/ncshipment/406951321983787352',
+            $links[0]['@attributes']['href']);
+        $this->assertEquals('self', $links[0]['@attributes']['rel']);
+        $this->assertEquals('application/vnd.cpc.ncshipment-v4+xml', $links[0]['@attributes']['media-type']);
```

Comment:
Line lengths standards please. Yes, coding standards apply to unit tests as well.